common/app/filters/RequestLoggingFilter.scala

```diff
@@ -29,3 +30,31 @@ object RequestLoggingFilter extends Filter with Logging with ExecutionContexts {
     result
   }
 }
+
+object DiscussionRequestLoggingFilter extends Filter with Logging with ExecutionContexts {
+  override def apply(next: (RequestHeader) => Future[Result])(rh: RequestHeader): Future[Result] = {
+
+    val requestId = Random.nextInt(Integer.MAX_VALUE)
+
+    if(Switches.LogAllDiscussionIncomingRequests.isSwitchedOn) {
+      log.info(s"Start handling ${rh.method} ${rh.uri} (requestID: ${requestId})")
+    }
+
+    val stopWatch = new StopWatch
+    val result = next(rh)
+    result onComplete {
+      case Success(response) =>
+        response.header.headers.get("X-Accel-Redirect") match {
+          case Some(internalRedirect) =>
+            log.info(s"${rh.method} ${rh.uri} took ${stopWatch.elapsed} ms and redirected to $internalRedirect (requestID: ${requestId})")
+
+          case None =>
+            log.info(s"${rh.method} ${rh.uri} took ${stopWatch.elapsed} ms and returned ${response.header.status} (requestID: ${requestId})")
+        }
+
+      case Failure(error) =>
+        log.warn(s"${rh.method} ${rh.uri} failed after ${stopWatch.elapsed} ms (requestID: ${requestId})", error)
+    }
+    result
+  }
+}
```

Comment:
Duplicating code so it applies only to the discussion app.
This is not supposed to be long living code. There is a switch that will force me to refactor back 👯